File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/dataset/TopSellingProducts.scala
package io.devorbit.sparklabs.batch.sales.dataset

import org.apache.spark.sql.functions.{col, count, desc}

object TopSellingProducts extends App {
  DatasetUtils.toDataFrame(DatasetUtils.orderItems)
    .join(DatasetUtils.toDataFrame(DatasetUtils.products), Seq("product_id"), "inner")
    .groupBy(col("product_id"))
    .agg(count("*").as("total_orders"))
    .orderBy(desc("total_orders"))
    .show(false)
}

File: r/sparklabs/batch/sales/sql/top_selling_products.R
source("r/sparklabs/batch/sales/sql/utils.R")

showDF(run_sql("
  SELECT oi.product_id, COUNT(*) AS total_orders
  FROM order_items oi
  JOIN products p ON oi.product_id = p.product_id
  GROUP BY oi.product_id
  ORDER BY total_orders DESC
"), truncate = FALSE)
